### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by panesofglass** [freya-fs/freya#104]
- [ ] Profile the framework using an application like [FreyaMusicStore](https://github.com/theimowski/FreyaMusicStore)
- [ ] Identify and eliminate hot spots
- [ ] Consider inlining

**Comment by panesofglass** [comment]
NOTE: this is almost a bug. I wonder whether anyone would use Freya in its current state.

**Comment by kolektiv** [comment]
What's a bug? I think I'm missing something :)

**Comment by panesofglass** [comment]
If the performance is so bad that people won't use the feature / platform, I would consider that a bug. At present, I think perf is only a feature, though probably a much needed one. Have you noticed any performance issues?

**Comment by kolektiv** [comment]
That's the thing, I haven't noticed any issues while I've been developing it. However, I haven't done any actual testing (going with the "correct first, then fast" approach). From the work I've done with it, it's always seemed reasonably snappy - the TodoBackend implementation doesn't feel noticeably slow, barring the usual .NET first request jazz.

Have we got some metrics or reports that performance is an issue? Definitely not saying we shouldn't profile and discuss anyway - that would be a de facto good thing. I'll take a proper look at the music store implementation now too, I think I might have some refactorings to suggest in places perhaps.

**Comment by theimowski** [comment]
wow, it turns out that when I move the definition of main pipeline to the `Project.Configuration`, the app startups 5 times faster or so. 

[here](https://github.com/theimowski/FreyaMusicStore/commit/dd602d9fcf5eb6c9419d581ed84e6496d82a04a6) is the change

**Comment by panesofglass** [comment]
That's odd.

**Comment by kolektiv** [comment]
Yeah it really is. Presumably that is just a change to application startup time, not running application performance? Without benchmarking yet, how does application performance "feel"?

**Comment by theimowski** [comment]
Yeah it was all down to the startup time - I was just complaining that I have to wait significant amount of time before the app started. 
The app performance when it's already running isn't bad (haven't done any benchmarks however)

**Comment by kolektiv** [comment]
That sounds positive. There's quite a lot that happens at startup, I'll
have to try benchmarking some of those aspects. I'd love to get some proper
repeatable benchmarks set up. It's good to hear that when it's running it
doesn't seem too bad on first impressions! I'm hoping it's going to be
fairly usable out of the box in that sense, although there's bound to be
some tweaks possible...
